chain/network/src/peer_manager/peer_manager_actor.rs

```diff
@@ -212,6 +209,75 @@ impl NetworkState {
         });
         PartialEdgeInfo::new(&self.config.node_id(), peer1, nonce, &self.config.node_key)
     }
+    /// Connects peer with given TcpStream.
+    /// This might fail if the other peers drop listener at its endpoint while establishing connection.
+    pub fn try_connect_stream(
+        self: &Arc<Self>,
+        clock: &time::Clock,
+        peer_manager_addr: Addr<PeerManagerActor>,
+        stream: TcpStream,
+        stream_cfg: StreamConfig,
+    ) {
+        // Start every peer actor on separate thread.
+        let arbiter = Arbiter::new();
+        let peer_cfg = match peer_actor::Config::new(stream, stream_cfg, None, self.clone()) {
+            Ok(cfg) => cfg,
+            Err(err) => {
+                warn!("peer_actor::Config::new(): {err}");
```

Comment:
it can fail for multiple reasons, in particular due to some system calls. I'm not sure which ones are expected and which ones are not right now. Degrading to tracing::into for now.